ecosystem/indexer-grpc/indexer-grpc-utils/src/file_store_operator_v2/file_store_reader.rs

```diff
@@ -118,13 +124,21 @@ impl FileStoreReader {
                 if num_to_skip > 0 {
                     transactions = transactions.split_off(num_to_skip);
                 }
+                let mut processed_range = (
+                    transactions.first().unwrap().version,
+                    transactions.last().unwrap().version,
+                );
+                if let Some(ending_version) = ending_version {
+                    transactions.retain(|t| t.version < ending_version);
+                    processed_range.1 = processed_range.1.min(ending_version - 1);
+                }
```

Comment:
If the transactions are in order, it'd probably be more performant to just take a slice from `transactions` instead of running `retain()` ?

Reply:
I was lazy :)... Changed to truncate.